CMakeLists.txt

```diff
@@ -1,75 +1,83 @@
+FetchContent_Declare(arcana.cpp
+    GIT_REPOSITORY https://github.com/microsoft/arcana.cpp.git
+    GIT_TAG f2757396e80bc4169f2ddb938ce25367a98ffdd0)
+FetchContent_Declare(arcore-android-sdk
+    GIT_REPOSITORY https://github.com/google-ar/arcore-android-sdk.git
+    GIT_TAG 98cb803de5482fb2b36effe8be3b5a0d3b726976)
 FetchContent_Declare(base-n
     GIT_REPOSITORY https://github.com/azawadzki/base-n.git
     GIT_TAG 7573e77c0b9b0e8a5fb63d96dbde212c921993b4)
+FetchContent_Declare(bgfx.cmake
+    GIT_REPOSITORY https://github.com/BabylonJS/bgfx.cmake.git
+    GIT_TAG 756234edf3105d6551fdfccd67570962d1641ef9)
+FetchContent_Declare(CMakeExtensions
+    GIT_REPOSITORY https://github.com/bghgary/CMakeExtensions.git
+    GIT_TAG 4d3ed5a16f3660ba0f144fb08b5021cf024d5d2d)
 FetchContent_Declare(glslang
     GIT_REPOSITORY https://github.com/BabylonJS/glslang.git
     GIT_TAG 6388a301b23edf0ae6a9342d46efa9cb21f05d9b)
-FetchContent_Declare(SPIRV-Cross
-    GIT_REPOSITORY https://github.com/BabylonJS/SPIRV-Cross.git
-    GIT_TAG 43e7999af1c9af7d3ed9d4fc3a44cf3e68235cfd)
-FetchContent_Declare(arcore-android-sdk
-    GIT_REPOSITORY https://github.com/google-ar/arcore-android-sdk.git
-    GIT_TAG 98cb803de5482fb2b36effe8be3b5a0d3b726976)
-FetchContent_Declare(OpenXR-SDK
-    GIT_REPOSITORY https://github.com/KhronosGroup/OpenXR-SDK.git
-    GIT_TAG 458984d7f59d1ae6dc1b597d94b02e4f7132eaba)
+FetchContent_Declare(googletest
+    URL "https://github.com/google/googletest/archive/refs/tags/v1.13.0.tar.gz")
+FetchContent_Declare(ios-cmake
+    GIT_REPOSITORY https://github.com/leetal/ios-cmake.git
+    GIT_TAG 4.4.1)
+FetchContent_Declare(JsRuntimeHost
+    GIT_REPOSITORY https://github.com/BabylonJS/JsRuntimeHost.git
+    GIT_TAG c147df420a5c7ca01a9c35e6962ab940e2a75912)
 FetchContent_Declare(OpenXR-MixedReality
     GIT_REPOSITORY https://github.com/microsoft/OpenXR-MixedReality.git
     GIT_TAG 67424511525b96a36847f2a96d689d99e5879503)
-FetchContent_Declare(bgfx
-    GIT_REPOSITORY https://github.com/BabylonJS/bgfx.cmake.git
-    GIT_TAG 3274a0b5755756d32cef18ed1859a342c1b90abd)
+FetchContent_Declare(OpenXR-SDK
+    GIT_REPOSITORY https://github.com/KhronosGroup/OpenXR-SDK.git
+    GIT_TAG 458984d7f59d1ae6dc1b597d94b02e4f7132eaba)
+FetchContent_Declare(SPIRV-Cross
+    GIT_REPOSITORY https://github.com/BabylonJS/SPIRV-Cross.git
+    GIT_TAG 43e7999af1c9af7d3ed9d4fc3a44cf3e68235cfd)
+# --------------------------------------------------
 
-set(CONTENT_TO_FETCH cmake-extensions arcana JsRuntimeHost base-n)
+FetchContent_MakeAvailable(CMakeExtensions)
 
 if(IOS)
-    FetchContent_MakeAvailable(ios-cmake)
+    FetchContent_MakeAvailable_With_Message(ios-cmake)
     set(CMAKE_TOOLCHAIN_FILE "${ios-cmake_SOURCE_DIR}/ios.toolchain.cmake" CACHE PATH "")
     set(PLATFORM "OS64COMBINED" CACHE STRING "")
-    set(DEPLOYMENT_TARGET "12" CACHE STRING "")
+    set(DEPLOYMENT_TARGET "13" CACHE STRING "")
     set(ENABLE_ARC OFF CACHE STRING "Enables or disables ARC support.")
```

Comment:
When this line is removed, there is this build error :
```
/Users/runner/work/1/s/buildiOS/_deps/bgfx.cmake-src/bgfx/src/renderer_mtl.h:983:5: error: ARC forbids explicit message send of 'release'
                                MTL_RELEASE(m_ptrMips[ii]);
                                ^           ~~~~~~~~~~~~~
/Users/runner/work/1/s/buildiOS/_deps/bgfx.cmake-src/bgfx/src/renderer_mtl.h:734:9: note: expanded from macro 'MTL_RELEASE'
```

Reply:
This is saying ARC should be off by default, and sub projects have to explicitly enable it if they want it, is that right?

Reply:
yes, per target, specify arc use with compilation flag `-fobjc-arc`